src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -133,7 +317,38 @@ public static ReadOnlySpan<char> TrimEnd(this ReadOnlySpan<char> span, char trim
         /// <remarks>If <paramref name="trimChars"/> is empty, white-space characters are removed instead.</remarks>
         public static ReadOnlySpan<char> Trim(this ReadOnlySpan<char> span, ReadOnlySpan<char> trimChars)
         {
-            return span.TrimStart(trimChars).TrimEnd(trimChars);
+            if (trimChars.IsEmpty)
```

Comment:
The fluent style was doing extra work. Expanding it allows for optimizations such as `for (; end >= start; end--)`

Reply:
Any perf numbers to justify the extra duplicated code?

Reply:
No it was just an eyeball optimization - the original code searches the entire span from the end. The new code only searches as far as the latest `start`:
Original: `for (; end >= 0; end--)`
Fixed: `for (; end >= start; end--)`
I can bench it if you want

Reply:
Reverted for now